xbout/plotting/utils.py

```diff
@@ -204,3 +204,77 @@ def _get_perp_vec(u1, u2, magnitude=0.04):
     v = np.linalg.norm((vx, vy))
     wx, wy = -vy / v * magnitude, vx / v * magnitude
     return wx, wy
+
+
+def _make_structured_triangulation(m, n):
+    # Construct a 'triangulation' of an (m x n) grid
+    # Returns an array of triples of (flattened) indices of the corners of the triangles
+    indices = np.zeros((m - 1, 2 * (n - 1), 3), dtype=np.uint32)
+    # indices[0] = [0, 1, n]
+    # indices[1] = [1, n+1, n]
+
+    # Each quadrilateral grid cell with corners (i,j), (i,j+1), (i+1,j+1), (i+1,j), at
+    # flattened index I=i*n+j is split into two triangles, one with corners (i,j),
+    # (i,j+1), (i+1,j) and the other with corners (i,j+1) (i+1,j+1), (i+1,j),
+
+    # Lower left triangles
+    # (i,j) corner
+    indices[:, ::2, 0] = (
+        n * np.arange(m - 1, dtype=np.uint32)[:, np.newaxis]
+        + np.arange((n - 1), dtype=np.uint32)[np.newaxis, :]
```

Comment:
(i, j) is always the same, so you could re-use that part :-)

But it is now much more readable anyways :+1: